php/post-types/class-block-post.php

```diff
@@ -1142,9 +1152,14 @@ public function list_table_content( $column, $post_id ) {
 			$block = new Block( $post_id );
 			echo esc_html( implode( ', ', $block->keywords ) );
 		}
-		if ( 'fields' === $column ) {
-			$block = new Block( $post_id );
-			echo esc_html( count( $block->fields ) );
+		if ( 'category' === $column ) {
+			global $post;
+
+			$block      = new Block( $post_id );
+			$categories = get_block_categories( $post );
```

Comment:
Would it be possible to pass `$post_id` instead of `$post` to `get_block_categories()`? This seems to work the same locally.
 
This might prevent using a `global $post`

Reply:
Yes, it is possible. In fact, passing `false` works also. In many ways, this is better, as it means that the list of block categories won't be filtered by third-parties.

`get_block_categories()` does expect a `WP_Post` object according to its docblock, but that parameter is only used as an additional argument in the `block_categories` filter. See [wp-admin/includes/post.php:2173](https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/post.php#L2173).

I very nearly just passed `false` instead of the global, but I decided not to because I felt it was a bit hacky. What do you think? Personally, I'd prefer to pass `false` if lean that way.

The thing that swayed my decision in the end was core's implementation here: [wp-admin/edit-form-blocks.php:90](https://github.com/WordPress/WordPress/blob/master/wp-admin/edit-form-blocks.php#L90)

Reply:
Hi @lukecarbis,
Good point, like you mentioned, [that Core implementation](https://github.com/WordPress/WordPress/blob/master/wp-admin/edit-form-blocks.php#L90) passes the global `$post` to `get_block_categories()`.

The way this PR uses the global `$post` isn't wrong, but I've usually seen globals declared at the top of the function. 

Maybe passing `get_post()` to `get_block_categories()` would be a little cleaner, but it'd probably work the same.

Reply:
>I very nearly just passed false instead of the global, but I decided not to because I felt it was a bit hacky. What do you think?

Yeah, it will probably work the same. 

But that probably wouldn't be my vote, as it violates the `@param WP_Post $post` tag for [get_block_categories](https://github.com/WordPress/WordPress/blob/c77e771c849a85b63cef1b9457ad1a501a331afa/wp-admin/includes/post.php#L2131), and the `@param WP_Post $post` tag for the [block categories filter](https://github.com/WordPress/WordPress/blob/c77e771c849a85b63cef1b9457ad1a501a331afa/wp-admin/includes/post.php#L2173)

Reply:
Great idea. Done in https://github.com/getblocklab/block-lab/pull/329/commits/be4a5b712215a29ddd63ce6850d31656f1b18b33.

I'd still be open to passing `false` or even `null` as an argument, if you think that would be better.

Reply:
Haha, our comments crossed paths.

Thanks, that commit looks good.